make sentinel class for  nbformat NO_CONVERT

Carreau opened the pull request:
Cf discussion in #7876 
Make Signature of nbconvert writes and a few others a little more sane.

Carreau committed:
# This is a combination of 2 commits.
# The first commit's message is:

make sentinel class for  nbformat NO_CONVERT

# This is the 2nd commit message:

make sentinel class for  nbformat NO_CONVERT

takluyver commented on jupyter_nbformat/__init__.py:
```diff
@@ -40,7 +40,17 @@ class NBFormatError(ValueError):
 # no-conversion singleton
-NO_CONVERT = object()
+class Sentinel(object):
+    """Prevent nbformat to convert notebooks to most recent version.
```
The docstring is really for the instance, not the class. Maybe it should have a second argument to take a description?

Carreau replied:
Done (push in progress)

minrk commented on IPython/html/static/components:
```diff
@@ -1 +1 @@
-Subproject commit 031e6c979ffd52c457a6ca8427116d9b2120e5b5
+Subproject commit e7f5a5a85f5dc0173e0b6ebacaa38c2af75a8195
```
Did you mean to update components?

Carreau committed:
make sentinel class for some kwargs.

The signature is a little more sane with
`fun(..., default=module.value)`
instead of
`fun(..., default=<object object at 0x12345678>)`

Carreau commented:
updated

```
In [2]: BaseFormatter.pop?
Signature: BaseFormatter.pop(self, typ, default=IPython.core.formatters._raise_key_error)
```

```
In [3]: from jupyter_nbformat import writes
In [4]:writes?
Signature: writes(nb, version=jupyter_nbformat.NO_CONVERT, **kwargs)
```

```
In [5]: from traitlets.traitlets import TraitType

In [6]: TraitType?
Init signature: TraitType(self, default_value=traitlets.traitlets.NoDefaultSpecified, allow_none=None, **metadata)
```

takluyver commented:
+1

Carreau commented:
Thanks !

On Thu, Apr 2, 2015 at 10:34 AM, Thomas Kluyver <notifications@github.com>
wrote:

> Merged #8192 <https://github.com/ipython/ipython/pull/8192>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/8192#event-271923935>.
>
